src/Pipeline.h

```diff
@@ -144,16 +151,30 @@ class Pipeline {
     /** Get the Funcs this pipeline outputs. */
     std::vector<Func> outputs() const;
 
-    /** Generate a schedule for the pipeline. */
-    //@{
+    /** Generate a schedule for the pipeline using the currently-default autoscheduler. */
     AutoSchedulerResults auto_schedule(const Target &target,
                                        const MachineParams &arch_params = MachineParams::generic());
-    //@}
 
-    /** Globally set the autoscheduler method to use whenever
-     * autoscheduling any Pipeline. Uses the built-in autoscheduler if
-     * passed nullptr. */
-    static void set_custom_auto_scheduler(AutoSchedulerFn auto_scheduler);
+    /** Generate a schedule for the pipeline using the specified autoscheduler. */
+    AutoSchedulerResults auto_schedule(const std::string &autoscheduler_name,
+                                       const Target &target,
+                                       const MachineParams &arch_params = MachineParams::generic());
+
+
+    /** Add a new the autoscheduler method with the given name. Does not affect the current default autoscheduler.
+     * It is an error to call this with the same name multiple times. */
+    static void add_autoscheduler(const std::string &autoscheduler_name, const AutoSchedulerFn autoscheduler);
+
+    /** Globally set the default autoscheduler method to use whenever
+     * autoscheduling any Pipeline when no name is specified. If the autoscheduler_name isn't in the
+     * current table of known autoschedulers, assert-fail.
+     *
+     * At this time, well-known autoschedulers include:
+     *  "Mullapudi2016" -- heuristics-based; the first working autoscheduler; currently built in to libHalide
+     *  "Adams2019"     -- aka "the ML autoscheduler"; currently located in apps/autoscheduler
```

Comment:
Maybe link to the paper: https://halide-lang.org/papers/autoscheduler2019.html
so people know what Adams2019 means.